pkg/scheduler/cache/cache.go

```diff
@@ -701,19 +725,9 @@ func (sc *SchedulerCache) BindPodGroup(job *schedulingapi.JobInfo, cluster strin
 	return nil
 }
 
-// GetPodVolumes get pod volume on the host
-func (sc *SchedulerCache) GetPodVolumes(task *schedulingapi.TaskInfo, node *v1.Node) (*volumescheduling.PodVolumes, error) {
-	return sc.VolumeBinder.GetPodVolumes(task, node)
-}
-
-// AllocateVolumes allocates volume on the host to the task
-func (sc *SchedulerCache) AllocateVolumes(task *schedulingapi.TaskInfo, hostname string, podVolumes *volumescheduling.PodVolumes) error {
-	return sc.VolumeBinder.AllocateVolumes(task, hostname, podVolumes)
-}
-
-// BindVolumes binds volumes to the task
-func (sc *SchedulerCache) BindVolumes(task *schedulingapi.TaskInfo, podVolumes *volumescheduling.PodVolumes) error {
-	return sc.VolumeBinder.BindVolumes(task, podVolumes)
+// SchedulerVolumeBinder returns volume binder
+func (sc *SchedulerCache) SchedulerVolumeBinder() VolumeBinder {
```

Comment:
Let's just add a `RevertVolumes` func for cache.

Reply:
further more, I'd like improve the steps as follows:

1. allocate remote resources (volumen, remote gpu pool...)
2. allocate local resource (cpu, mem, gpu ...)
3. discard if any error

Which means we should not enhance this case by case :) Anyway, I'm ok to get this merged firstly to handle the issue, and open a seperate PR to do the refactor.

Reply:
> Let's just add a `RevertVolumes` func for cache.

My thought is cache‘s get/allocate/bind method is just a proxy for VolumeBinder and VolumeBinder is also public interface, returns volumebinder may simplify the code. 

I will update it for that.

Reply:
> Let's just add a `RevertVolumes` func for cache.

updated